packages/omgx/wallet-frontend/src/services/networkService.js

```diff
@@ -1439,12 +1451,46 @@ class NetworkService {
         this.provider.getSigner()
       )
 
-      const approveStatus = await ERC20Contract.approve(
-        approveContractAddress,
-        value_Wei_String
+      /***********************/
+
+      let allowance_BN = await ERC20Contract.allowance(
+        this.account,
+        approveContractAddress
+      ) 
+      console.log("Allowance:",allowance_BN)
+
+      /* OMG IS A SPECIAL CASE - allowance needs to be 
+      set to zero, and then set to actual amount */
+      if( allowance_BN.gt(BigNumber.from(0)) && 
+          (currency.toLowerCase() === this["L1_OMG_Address"].toLowerCase())
+      ) 
+      {
+        console.log("OMG Token allowance reset")
```

Comment:
for cases when the existing allowance is already greater than the amount to deposit, should we skip the allowance reset?